linkml/generators/markdowngen.py

```diff
@@ -296,7 +296,15 @@ def element_header(self, obj: Element, name: str, curie: str, uri: str) -> None:
             obj_type = 'Subset'
         else:
             obj_type = 'Class'
-        self.header(1, f"{obj_type}: {name}" + (f" _(deprecated)_" if obj.deprecated else ""))
+
+        # TODO: for some reason obj.deprecated dosn't appear to be True even
```

Comment:
For some reason, even though I added `deprecated` as a `slot` in [`kitchen_sink.yaml`](https://github.com/linkml/linkml/blob/markdown-gen-updates/tests/test_generators/input/kitchen_sink.yaml#L65), `obj.deprecated` was returning `None` instead of `True`, which I did not expect.

I'm assuming this is either because:
a. Adding 'deprecated' as a slot is not how we mark a class as deprecated, or 
b. There is some error which is causing the 'deprecated' slot this to not be detected.

If (b), I assume the issue is in [`linkml_runtime/linkml_model/meta`](https://github.com/linkml/linkml-runtime/blob/main/linkml_runtime/linkml_model/meta.py), but I'm not sure exactly where or how to fix it yet. It looks like the fix would need to be in several different classes (which may mean that the way inheritance and mixins are setup in these classes are not optimally). 

So for now, as a hack, I've detected 'deprecated' another way. But I'd like to hear back concerning this. If it is indeed a problem in `meta.py`, I can try to tackle, or make a new issue for it.

Reply:
So the range of deprecated is a string, containing the deprecation reason. This is admittedly very odd! We can discussion tomorrow, but I suspect this may be the heart of your issue

Reply:
Hmm, I see. Well, I do have the `range` defined, though it does just say "string":
```yml
  deprecated:
    name: deprecated
    definition_uri: https://w3id.org/linkml/deprecated
    description: Description of why and when this element will no longer be used
    from_schema: https://w3id.org/linkml/meta
#    domain: God
    range: string
    slot_uri: linkml:deprecated
    owner: deprecated
    domain_of:
    - common_metadata
```

will discuss later

Reply:
I think this was caused, as we discovered, because I declared deprecated under 'slots', rather than as a root attribute of the class.